integration/monai/executors/client_algo_executor.py

```diff
@@ -0,0 +1,213 @@
+        validate_result = DXO(
+            data_kind=DataKind.METRICS, data={MetaKey.INITIAL_METRICS: test_key_metric}
+        ).to_shareable()
+
+        self.client_algo.train(exchangeobj_from_shareable(shareable), extra={ExtraItems.ABORT: abort_signal})
+        local_weights_eo = weights_to_numpy(self.client_algo.get_weights())
+
+        # check returned weights dict
+        if local_weights_eo.weights is None:
+            self.log_error(fl_ctx, "Returned exchange object doesn't contain weights.")
+            return make_reply(ReturnCode.ERROR)
+
+        if local_weights_eo.weight_type == WeightType.WEIGHTS:
+            data_kind = DataKind.WEIGHTS
+        elif local_weights_eo.weight_type == WeightType.WEIGHT_DIFF:
+            data_kind = DataKind.WEIGHT_DIFF
+        else:
+            self.log_error(
+                fl_ctx,
+                f"Returned `WeightType` not supported. Expected {WeightType.WEIGHTS} or {WeightType.WEIGHT_DIFF} but got {local_weights_eo.get_weight_type()}",
+            )
+            return make_reply(ReturnCode.ERROR)
+
+        # get the number of executed steps
+        statistics = local_weights_eo.statistics
+        executed_steps = statistics.get(FlStatistics.NUM_EXECUTED_ITERATIONS)
+        if executed_steps:
+            meta = {MetaKey.NUM_STEPS_CURRENT_ROUND: executed_steps}
+        else:
+            meta = None
+
+        # TODO: support returned optimizer state (DXO collection?)
+        train_result = DXO(data_kind=data_kind, data=local_weights_eo.weights, meta=meta).to_shareable()
+
+        if not (train_result and isinstance(train_result, Shareable)):
+            return make_reply(ReturnCode.EMPTY_RESULT)
+
+        # if the client_algo returned the valid BEFORE_TRAIN_VALIDATE result, set the INITIAL_METRICS in
+        # the train result, which can be used for best model selection.
+        if (
+            validate_result
+            and isinstance(validate_result, Shareable)
+            and validate_result.get_return_code() == ReturnCode.OK
+        ):
+            try:
+                metrics_dxo = from_shareable(validate_result)
+                train_dxo = from_shareable(train_result)
+                train_dxo.meta[MetaKey.INITIAL_METRICS] = metrics_dxo.data.get(MetaKey.INITIAL_METRICS, 0)
+                return train_dxo.to_shareable()
+            except ValueError:
+                return train_result
+        else:
+            return train_result
+
+    def submit_model(self, shareable: Shareable, fl_ctx: FLContext) -> Shareable:
+        model_name = shareable.get_header(AppConstants.SUBMIT_MODEL_NAME)
+        local_weights = weights_to_numpy(self.client_algo.get_weights(extra={ExtraItems.MODEL_NAME: model_name}))
+        submit_model_result = DXO(data_kind=DataKind.WEIGHTS, data=local_weights).to_shareable()
+        if submit_model_result and isinstance(submit_model_result, Shareable):
+            return submit_model_result
```

Comment:
``if submit_mode_result and isinstance(submit_model_result, Shareable):`` this always evaluate to True?

Reply:
I'm following the same logic as in https://github.com/NVIDIA/NVFlare/blob/dev/nvflare/app_common/executors/learner_executor.py.